src/runtime/os_darwin.go

```diff
@@ -229,3 +229,38 @@ func call_syscall6(fn, a1, a2, a3, a4, a5, a6 uintptr) int32
 
 //export tinygo_syscall6X
 func call_syscall6X(fn, a1, a2, a3, a4, a5, a6 uintptr) uintptr
+
+//go:linkname os_runtime_executable_path os.runtime_executable_path
+func os_runtime_executable_path() string {
+	argv := (*unsafe.Pointer)(unsafe.Pointer(main_argv))
+
+	// skip over argv
+	argv = (*unsafe.Pointer)(unsafe.Add(unsafe.Pointer(argv), (uintptr(main_argc)+1)*unsafe.Sizeof(argv)))
+
+	// skip over envv
+	for (*argv) != nil {
+		argv = (*unsafe.Pointer)(unsafe.Add(unsafe.Pointer(argv), unsafe.Sizeof(argv)))
+	}
+
+	// next string is exe path
+	argv = (*unsafe.Pointer)(unsafe.Add(unsafe.Pointer(argv), unsafe.Sizeof(argv)))
+
+	cstr := unsafe.Pointer(*argv)
+	length := strlen(cstr)
+	argString := _string{
+		length: length,
+		ptr:    (*byte)(cstr),
+	}
+	executablePath := *(*string)(unsafe.Pointer(&argString))
```

Comment:
Why not `executablePath := unsafe.String((*byte)(cstr), length)`?

Reply:
We still support Go 1.19, and unsafe.String was introduced in Go 1.20.

Reply:
We don't (yet) use `unsafe.String()` in the runtime package, although there is a comment to that effect: https://github.com/tinygo-org/tinygo/blob/3e60eeb368f25f237a512e7553fd6d70f36dc74c/src/runtime/runtime.go#L114

Reply:
I see, thanks. I *did* in fact look up when `unsafe.String` was introduced but then went ahead and assumed 1.20 was more than safe. Out of curiosity, is there any reason for keeping support for such a old version of Go?

Reply:
We could indeed drop support for this older version by now. One consideration might be that Debian bookworm (current stable) uses Go 1.19, dropping support for that means people will need to upgrade the Go version on Debian.